### ISSUE TITLE
Create Threat Intelligence apps

### ISSUE THREAD

**Issue by frikky** [frikky/Shuffle-apps#24]
Using the App creator, OpenAPI or Python directly, create any of the following apps (or other ones!):

MISP
Passivetotal
Recorded Future
Secureworks
Shodan
Virustotal
IBM xforce
OpenCTI
ATP
Fireeye

**Comment by frikky** [comment]
More sources (OpenCTI): https://luatix.notion.site/OpenCTI-Ecosystem-868329e9fb734fca89692b2ed6087e76

Extra: Create a simple workflow that merges threat lists daily into the shuffle K:V store.

**Comment by cvdsouza** [comment]
It would be pretty useful to see an app for OpenCTI in Shuffle that can be used for enrichment of data within Shuffle as well using shuffle to push data into OpenCTI. Similar to Shuffle this platform has been fast maturing and is a very easily adoptable opensource threat intelligence platform.

**Comment by frikky** [comment]
@cvdsouza agreed! If you or someone else that uses OpenCTI would be willing to work with us to build it out, we can set it up and prepare everything for OpenCTI very easily :)

**Comment by weslambert** [comment]
I might be able to help with putting something to together. Let me know!

**Comment by frikky** [comment]
Yes please Wes! We still got some work to do with Velociraptor, and doing some at the intersection of the two would be even better

**Comment by cvdsouza** [comment]
awesome. thank you both , really appreciate it. 
OpenCTI has a python client that is updated to stay compliant with the platform updates that are released : https://github.com/OpenCTI-Platform/client-python

There is a Demo instance of OpenCTI that is always open to the public to test against : https://demo.opencti.io/dashboard
I also have a research instance of OpenCTI that I recently built up a couple of weeks ago, so if the demo version doesn't work, I'd be happy to share access as well. 

As for use-cases, the ones that I've used with XSOAR that I think would be beneficial for Shuffle would be : 
- Get/Search Indicator
- Get/Search Observable
- Create Indicator
- Update Indicator 
- Organization create
- Organization List
- Label create
- Label list
- External Reference create

**Comment by azgaviperr** [comment]
My OpenCTI Stack using docker swarm will be available soon also.

**Comment by frikky** [comment]
Update: Good initial testing. There's a couple issues with GraphQL and the use of dollar signs & same URL that screw up things, so we're trying to fix that for Shuffle itself. OpenAPI wasn't meant to do this sort of stuff, so we got to bend the rules~ :+1: 

![image](https://user-images.githubusercontent.com/5719530/195071853-afd6d90a-596b-405f-b0a6-51ba0cab9d5f.png)

**Comment by frikky** [comment]
Aaand edit 2: We pushed the platform fixes, and it's been deployed with a base set of actions.

Have a look here:
https://shuffler.io/apps/24555182e0063c1800d0c8e320e0892a

It's all from reversing the UI's interactions and can probably be optimized quite a bit. @weslambert - I'd love if you could take over some of this work :)
